File: server/src/main/java/com/server/service/IdeaService.java
package com.server.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.server.bean.Idea;
import com.server.repo.IdeaRepo;

import java.util.List;
import java.util.Optional;

@Service
public class IdeaService {

	private IdeaRepo ideaRepository;

	@Autowired
	public IdeaService(IdeaRepo ideaRepository) {
		this.ideaRepository = ideaRepository;
	}

	public List<Idea> getAllIdeas() {
		return ideaRepository.findAll();
	}

	public Idea getIdeaById(String id) {
		Optional<Idea> optionalIdea = ideaRepository.findById(id);
		return optionalIdea.orElse(null);
	}

	public Idea addIdea(Idea idea) {
		return ideaRepository.save(idea);
	}

	public Idea updateIdea(String id, Idea idea) {
		idea.setId(id);
		return ideaRepository.save(idea);
	}

	public void deleteIdea(String id) {
		ideaRepository.deleteById(id);
	}

	public void updateSuggestionAndStatus(String ideaId, String newSuggestion, String newStatus) {
		Optional<Idea> optionalIdea = ideaRepository.findById(ideaId);

		if (optionalIdea.isPresent()) {
			Idea idea = optionalIdea.get();
			idea.updateSuggestionAndStatus(newSuggestion, newStatus);
			ideaRepository.save(idea);
		} else {
			System.out.println("Idea not found!");
		}
	}

}

File: client/src/service/idea.service.js
import { useFetchWrapper } from "../utils/fetchWrapper";

export const IdeaService = () => {
  const fetchWrapper = useFetchWrapper();
  const baseURL = "/ideas";

  const getAllIdeas = () => {
    const url = `${baseURL}/all`;
    return fetchWrapper.get(url);
  };

  const getIdeaById = (id) => {
    const url = `${baseURL}/${id}`;
    return fetchWrapper.get(url);
  };

  const addIdea = (idea) => {
    const url = `${baseURL}/add`;
    return fetchWrapper.post(url, idea);
  };

  const updateIdea = (id, idea) => {
    const url = `${baseURL}/update/${id}`;
    return fetchWrapper.put(url, idea);
  };

  const deleteIdea = (id) => {
    const url = `${baseURL}/delete/${id}`;
    return fetchWrapper.delete(url);
  };

  return { getAllIdeas, getIdeaById, addIdea, updateIdea, deleteIdea };
};
